homeassistant/components/neato/vacuum.py

```diff
@@ -35,15 +40,57 @@
 ATTR_CLEAN_SUSP_COUNT = 'clean_suspension_count'
 ATTR_CLEAN_SUSP_TIME = 'clean_suspension_time'
 
+ATTR_MODE = 'mode'
+ATTR_NAVIGATION = 'navigation'
+ATTR_CATEGORY = 'category'
+ATTR_ZONE = 'zone'
+
+SERVICE_NEATO_CUSTOM_CLEANING = 'neato_custom_cleaning'
+
+SERVICE_NEATO_CUSTOM_CLEANING_SCHEMA = VACUUM_SERVICE_SCHEMA.extend({
+    vol.Optional(ATTR_MODE, default=2): cv.positive_int,
+    vol.Optional(ATTR_NAVIGATION, default=1): cv.positive_int,
+    vol.Optional(ATTR_CATEGORY, default=4): cv.positive_int,
+    vol.Optional(ATTR_ZONE): cv.string
+})
+
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Neato vacuum."""
     dev = []
     for robot in hass.data[NEATO_ROBOTS]:
         dev.append(NeatoConnectedVacuum(hass, robot))
+
     _LOGGER.debug("Adding vacuums %s", dev)
     add_entities(dev, True)
 
+    if dev:
```

Comment:
Invert this check and return if true. Also make it earlier, before the call to `add_entities`.

Reply:
Done